### PR TITLE
feat: allow addition of trade using quote amount instead of rate

### LINKED ISSUE #2446: Allow addition of trade using quote amount instead of rate
You can only specify the rate for a trade. I would like the option for the rate to be calculated based on the quote amount total.

Note: This is how delta works and how the data is displayed when exporting to CSV from delta. Without this option, it creates rounding errors in the report.

For example, if I spend 1001.23 EUR to buy some BTC I would like that information to display in the "Paid in Asset" column of the report accurately. Right now it uses the rate to calculate it, which leads to rounding errors (values with many decimal places). This is because I have to calculate the rate and enter that into rotki, which is then used to convert back to "Paid in Asset".

### PR DESCRIPTION
**Pull request by lukicenturi**
Closes #2446

## Checklist

- [ ] The PR modified the frontend, and updated the [user guide](https://github.com/rotki/rotki/blob/develop/docs/usage_guide.rst) to reflect the changes.
- [x] In trade addition form, user can choose to input `rate` or `quoteAmount` instead.
image

### CODE REVIEW

File: frontend/app/src/components/ExternalTradeForm.vue [hunk 1]
```diff
@@ -281,6 +322,34 @@ export default class ExternalTradeForm extends Vue {
...
+  async onAmountChange() {
+    this.onRateChange();
+    this.onQuoteAmountChange();
+  }
+
+  @Watch('rate')
+  onRateChange() {
+    if (this.amount && this.rate && this.selectedCalculationInput === 'rate') {
+      this.quoteAmount = bigNumberify(
+        parseFloat(this.amount) * parseFloat(this.rate)
```
Comment (kelsos) [inline frontend/app/src/components/ExternalTradeForm.vue:335]: just fyi, you can also make BigNumbers out of the properties and perform this operation between bignumbers the library used is [bignumber.js/](https://github.com/MikeMcl/bignumber.js/)

**Comment by kelsos** [comment]
@lukicenturi it seems that the e2e tests for adding trades failed.

So one request would be to also add a new e2e test that adds a trade by using the quote amount.

The second thing is, do you think we could move the switch button to a column at the end, and the text fields to another so that there is not so much empty space between them?

File: frontend/app/src/components/ExternalTradeForm.vue (hunk 1 above)
Comment (lukicenturi) [reply frontend/app/src/components/ExternalTradeForm.vue:335]: Okay, will change it

**Review by kelsos — APPROVED** [review]
lgtm, the component looks really nice now :+1:

**Pull request merged**